### ISSUE TITLE
Feature: rename opam switches

### ISSUE THREAD

**Issue by yomimono** [ocaml/opam#2609]
I have a workflow where I make fresh switch aliases to create separate environments where I've pinned a particular set of branches.  Occasionally I want to rename them to resolve an ambiguity or reflect some set of things which has been merged.

**Comment by AltGr** [comment]
This shouldn't be difficult, but in practice there are two issues:
1. opam binds the switch name to the directory name `~/.opam/<switch-name>/`
2. OCaml doesn't handle being relocated very well

So this would require adding the target directory in the list of switches in `~/.opam/config`. While there is no big trouble in this, many have come to assume that `opam config var prefix` is `opamroot/<switch-name>`. Maybe something we want anyway ?

**Comment by Drup** [comment]
I would like that feature as well, for similar reasons than @yomimono. Especially when I have some custom ocaml compiler and later realize that I would like to give a proper name to the switch, instead of the auto-generated one (pr, compiler conf, etc).

**Comment by AltGr** [comment]
Not really what you're asking for, but switches now have a short description that is editable.

**Comment by brettgilio** [comment]
What is the state of this issue?

**Comment by dra27** [comment]
TL;DR it’s not being directly worked on and it won’t land any time soon.

The more optimistic version: 🙂
- I am gradually working on changes to OCaml which will permit the standard library to a new location. This is the largest piece in the puzzle for being able, in general, to move the compiler
- others are working on reproducible builds in general, and the work needed for that adds features and hardens package build systems to allow moving installed artefacts
- there are halfway solutions - @AltGr mentions one above (it has the weakness that you can’t reuse the original name of the switch); it’s possible to set the tree up so that OCaml remains in the same place and you use `ocamlfind` to select between different ‘switches’ of packages, but `opam` has no direct support for this kind of workflow (the cross-compiler repos use this `ocamlfind` feature); it’s also possible to move the compiler if you configure without shebang scripts or use native compilation only and then always set the `OCAMLLIB` environment variable, but it’s a very brittle setup

I’m not sure anyone wants it enough to be implementing the halfway solutions, but once the compiler is relocatable, that will be used quickly for caching, and adding switch renaming will be a relatively easy change to make (and test)
